cpp/src/parquet/arrow/test_util.h

```diff
@@ -131,8 +131,12 @@ ::arrow::enable_if_fixed_size_binary<ArrowType, Status> NonNullArray(
 
 static void random_decimals(int64_t n, uint32_t seed, int32_t precision, uint8_t* out) {
   std::default_random_engine gen(seed);
-  std::uniform_int_distribution<uint32_t> d(0, std::numeric_limits<uint8_t>::max());
-  const int32_t required_bytes = ::arrow::DecimalType::DecimalSize(precision);
+  // Generate smaller values than requested to avoid generating invalid decimals for the
+  // given precision
+  const int32_t required_bytes =
+      std::max(1, ::arrow::DecimalType::DecimalSize(precision) - 1);
+  std::uniform_int_distribution<uint32_t> d(
+      0, required_bytes <= 1 ? 9 : std::numeric_limits<uint8_t>::max());
```

Comment:
Same comment as in the other occurrence, though this may be less trivial in the Parquet codebase.